symengine/parser.cpp

```diff
@@ -626,9 +636,13 @@ class ExpressionParser
     }
 };
 
-RCP<const Basic> parse(const std::string &s)
+RCP<const Basic> parse(const std::string &s, bool convert_xor)
 {
     ExpressionParser p;
+    if (convert_xor) {
+        p.op_precedence["^"] = p.op_precedence["**"];
+    }
+    p.convert_xor = convert_xor;
```

Comment:
It's better if the above 4 lines are moved to the constructor of `ExpressionParser`